examples/platform/nrfconnect/util/QRCodeUtil.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "QRCodeUtil.h"
+
+#include <platform/CHIPDeviceLayer.h>
+
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
+
+#include <logging/log.h>
```

Comment:
if this file used CHIP logging instead of nrf logging, this facility could be moved into CHIP?

Reply:
Since there's currently no application component within CHIP, "moved into CHIP" you mean e.g. examples/common/chip-app-server?

Reply:
No, I suggest moving this to src/setup_payload and including it in the library built there.

examples/common/ really only pertains to code that shouldn't end up in CHIP, like that used to deal with displays, etc.

Reply:
I'm not sure.... by moving this code to setup_payload we would introduce a dependency between setup_payload and the platform library and I wonder if what we could potentially benefit from such a change is worth it. I would rather see code which glues underlying CHIP components together in the CHIP "application server" library which is to be created in #2927. What do you think?
For now I think I can remove non-essential parts of the PR which were only meant to address code duplication until #2927 is finished.

Reply:
excellent points